### PR TITLE
Split YUV2RGB shader into GL/GLES

### PR DESCRIPTION
**Pull request by lrusak**
Tested on GL and GLES this works the same way as before.

This cleans up much of the ifdefs in the code.

Should be tested on windows as this affects it and I don't build for windows.

### CODE REVIEW

**Review by FernetMenta — APPROVED** [review]

File: xbmc/cores/VideoPlayer/VideoRenderers/VideoShaders/YUV2RGBShaderGLES.h [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+   ~BaseYUV2RGBGLSLShader() override;
+    void SetField(int field) override { m_field  = field; }
+    void SetWidth(int w) override { m_width  = w; }
+    void SetHeight(int h) override { m_height = h; }
+
+    void SetBlack(float black) override { m_black    = black; }
+    void SetContrast(float contrast) override { m_contrast = contrast; }
+    void SetNonLinStretch(float stretch) override { m_stretch = stretch; }
+
+    virtual GLint GetVertexLoc() { return m_hVertex; }
```
Comment (Rechi) [inline xbmc/cores/VideoPlayer/VideoRenderers/VideoShaders/YUV2RGBShaderGLES.h]: mark this function override

File: xbmc/cores/VideoPlayer/VideoRenderers/VideoShaders/YUV2RGBShaderGLES.h [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    virtual GLint GetYcoordLoc() { return m_hYcoord; }
```
Comment (Rechi) [inline xbmc/cores/VideoPlayer/VideoRenderers/VideoShaders/YUV2RGBShaderGLES.h]: mark this function override

File: xbmc/cores/VideoPlayer/VideoRenderers/VideoShaders/YUV2RGBShaderGLES.h [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    virtual GLint GetUcoordLoc() { return m_hUcoord; }
```
Comment (Rechi) [inline xbmc/cores/VideoPlayer/VideoRenderers/VideoShaders/YUV2RGBShaderGLES.h]: mark this function override

File: xbmc/cores/VideoPlayer/VideoRenderers/VideoShaders/YUV2RGBShaderGLES.h [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    virtual GLint GetVcoordLoc() { return m_hVcoord; }
```
Comment (Rechi) [inline xbmc/cores/VideoPlayer/VideoRenderers/VideoShaders/YUV2RGBShaderGLES.h]: mark this function override

File: xbmc/cores/VideoPlayer/VideoRenderers/VideoShaders/YUV2RGBShaderGLES.h [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+    virtual void SetMatrices(GLfloat *p, GLfloat *m) { m_proj = p; m_model = m; }
```
Comment (Rechi) [inline xbmc/cores/VideoPlayer/VideoRenderers/VideoShaders/YUV2RGBShaderGLES.h]: mark this function override

File: xbmc/cores/VideoPlayer/VideoRenderers/VideoShaders/YUV2RGBShaderGLES.h [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+    virtual void SetAlpha(GLfloat alpha) { m_alpha = alpha; }
```
Comment (Rechi) [inline xbmc/cores/VideoPlayer/VideoRenderers/VideoShaders/YUV2RGBShaderGLES.h]: mark this function override

**Review by Rechi — COMMENTED** [review]
a few missing overrides discovered by jenkins

**Review by afedchin — APPROVED** [review]

**Comment by lrusak** [comment]
jenkins build and merge

**Pull request merged**